src/bluesky/callbacks/tiled_writer.py

```diff
@@ -301,21 +112,57 @@ def descriptor(self, doc: EventDescriptor):
             desc_node = self.root_node.create_container(key=desc_name, metadata=metadata)
             desc_node.create_container(key="external")
             desc_node.create_container(key="internal")
+            desc_node.create_container(key="config")
         else:
             # Get existing descriptor node (with fixed and variable metadata saved before)
             desc_node = self.root_node[desc_name]
 
         # Update (add new values to) variable fields of the metadata
         metadata = deep_update(dict(desc_node.metadata), var_fields)
         desc_node.update_metadata(metadata)
+
+        # Keep specifications for external and internal data_keys for faster access
+        self.data_keys_int.update({k: v for k, v in metadata["data_keys"].items() if "external" not in v.keys()})
+        self.data_keys_ext.update({k: v for k, v in metadata["data_keys"].items() if "external" in v.keys()})
+
+        # Write the configuration data: loop over all detectors
+        conf_node = desc_node["config"]
+        for det_name, det_dict in conf_dict[uid].items():
+            print(det_name, det_dict)
+            df_dict = {"descriptor_uid": uid}
+            df_dict.update(det_dict.get("data", {}))
+            df_dict.update({f"ts_{c}": v for c, v in det_dict.get("timestamps", {}).items()})
+            df = pd.Series(df_dict).to_frame().T
+            if det_name in conf_node.keys():
+                conf_node[det_name].append_partition(df, 0)
+            else:
+                conf_node.new(
+                    structure_family=StructureFamily.table,
+                    data_sources=[
+                        DataSource(
+                            structure_family=StructureFamily.table,
+                            structure=TableStructure.from_pandas(df),
+                            mimetype="text/csv",
+                        ),
+                    ],
+                    key=det_name,
+                    metadata=det_dict["data_keys"],
+                )
+                conf_node[det_name].write_partition(df, 0)
+
         self._desc_nodes[uid] = desc_node
 
     def event(self, doc):
-        descriptor_node = self._desc_nodes[doc["descriptor"]]
-        parent_node = descriptor_node["internal"]
-        df_dict = {c: [v] for c, v in doc["data"].items()}
-        df_dict.update({f"ts_{c}": [v] for c, v in doc["timestamps"].items()})
-        df = pd.DataFrame(df_dict)
+        desc_node = self._desc_nodes[doc["descriptor"]]
+
+        # Process _internal_ data -- those without external flag or those that have been filled
+        data_keys_spec = {k: v for k, v in self.data_keys_int.items() if doc["filled"].get(k, True)}
+        data_keys_spec.update({k: v for k, v in self.data_keys_ext.items() if doc["filled"].get(k, False)})
```

Comment:
It will, but only if the 'external' fields have been "filled" (i.e. the data is passed in the document directly, not by reference). In this case, the key should be processed as internal (without creating references).
It's probably unnecessary to check ` if doc["filled"].get(k, True)` in line 159 though, i.e. internal keys will never be "filled" if I understand it correctly, but I think i included it there just in case.